### PR TITLE
ci: run fat container docker inside perf tests and update setup-perf-test workflow to use localhost for accessing dev environment

### PR DESCRIPTION
**Pull request by Rhitottam**

### CODE REVIEW

File: .github/workflows/perf-test.yml [hunk 1]
```diff
@@ -183,7 +183,13 @@ jobs:
           docker run --name test-event-driver -d -p 2222:22 -p 5001:5001 -p 3306:3306 \
           -p 5432:5432 -p 28017:27017 -p 25:25 --privileged --pid=host --ipc=host --volume /:/host -v ~/git-server/keys:/git-server/keys \
           -v ~/git-server/repos:/git-server/repos  appsmith/test-event-driver:latest
-          
+          cd cicontainerlocal
```
[the remark this answers is not in the document]
Comment (Aishwarya-U-R) [reply .github/workflows/perf-test.yml:186]: cicontainerlocal - also should be created.
`mkdir -p cicontainerlocal/stacks/configuration/` - reference from integrations yml file

**Review by kaustavy2k — APPROVED** [review]

File: .github/workflows/perf-test.yml [hunk 2]
```diff
@@ -171,14 +171,40 @@ jobs:
...
+      - name: Download S3 image
+        uses: keithweaver/aws-s3-github-action@v1.0.0
+        with:
+          command: cp
+          source: s3://ci-assets--appsmith/
+          destination: /home/runner/work/appsmith/appsmith/cicontainerlocal/oldstack
+          aws_access_key_id: ${{ secrets.S3_CI_ASSETS_ACCESS_KEY_ID }}
+          aws_secret_access_key: ${{ secrets.S3_CI_ASSETS_SECRET_ACCESS_KEY }}
+          aws_region: ap-south-1
+          flags: --recursive
```
Comment (Aishwarya-U-R) [inline .github/workflows/perf-test.yml:200]: Downloading S3 image is not required for perf tests

File: .github/workflows/perf-test.yml [hunk 3]
```diff
@@ -220,8 +254,8 @@ jobs:
           APPSMITH_GOOGLE_MAPS_API_KEY: ${{ secrets.APPSMITH_GOOGLE_MAPS_API_KEY }}
           POSTGRES_PASSWORD: postgres
         run: |
-          chmod a+x ./perf/setup-perf-test.sh
-          ./perf/setup-perf-test.sh
+          chmod a+x ./cypress/setup-test-ci.sh
+          ./cypress/setup-test-ci.sh
```
Comment (SatishGandham) [inline .github/workflows/perf-test.yml:258]: Lets maintain our own setup script.

File: app/client/perf/setup-perf-test.sh [hunk 4]
```diff
@@ -5,39 +5,21 @@
...
-# sudo mkdir -p git-server/keys
-# sudo mkdir -p git-server/repos
-
-# sudo docker run --name TED -d -p 2222:22 -p 5001:5001 -p 3306:3306 \
-#   -p 5432:5432 -p 28017:27017 -p 25:25 --privileged --pid=host --ipc=host --volume /:/host -v ~/git-server/keys:/git-server/keys \
-#   -v ~/git-server/repos:/git-server/repos  appsmith/test-event-driver:latest
-
-# echo "Waiting for test event driver to start"
-# sleep 50
+sleep 30
```
Comment (SatishGandham) [inline app/client/perf/setup-perf-test.sh:22]: Can we remove this sleep?

File: .github/workflows/perf-test.yml [hunk 5]
```diff
@@ -244,16 +278,16 @@ jobs:
...
-        if: steps.run_result.outputs.run_result != 'success'
-        run: |
-          if lsof -i :8080; then
-            echo "Server Found"
-          else
-            echo "Server Not Started. Printing logs from server process"
-            cat app/server/nohup.out
-            exit 1
-          fi
+      # - name: Exit if Server hasnt started
```
Comment (SatishGandham) [inline .github/workflows/perf-test.yml:281]: Is this no longer required?

**Review by SatishGandham — CHANGES_REQUESTED** [review]

**Review by SatishGandham — APPROVED** [review]

**Pull request merged**